app/code/Magento/InventoryCatalogSearch/Test/Integration/Plugin/Adapter/Mysql/Aggregation/DataProvider/StockConditionJoinerTest.php

```diff
@@ -0,0 +1,106 @@
+<?php
+/**
+ * Copyright :copyright: Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryCatalogSearch\Test\Integration\Plugin\Adapter\Mysql\Aggregation\DataProvider;
+
+use Magento\CatalogSearch\Model\Adapter\Mysql\Aggregation\DataProvider\SelectBuilderForAttribute\StockConditionJoiner;
+use Magento\Framework\App\ResourceConnection;
+use Magento\Store\Model\StoreManagerInterface;
+use Magento\TestFramework\Helper\Bootstrap;
+use PHPUnit\Framework\TestCase;
+
+class StockConditionJoinerTest extends TestCase
+{
+    /**
+     * @var StockConditionJoiner
+     */
+    private $stockConditionJoiner;
+
+    /**
+     * @var ResourceConnection
+     */
+    private $resource;
+
+    /**
+     * @var StoreManagerInterface
+     */
+    private $storeManager;
+
+    /**
+     * @var string
+     */
+    private $storeCodeBefore;
+
+    /**
+     * @inheritdoc
+     */
+    protected function setUp()
+    {
+        $this->resource = Bootstrap::getObjectManager()->get(ResourceConnection::class);
+        $this->stockConditionJoiner = Bootstrap::getObjectManager()->get(StockConditionJoiner::class);
+        $this->storeManager = Bootstrap::getObjectManager()->get(StoreManagerInterface::class);
+        $this->storeCodeBefore = $this->storeManager->getStore()->getCode();
+
+        parent::setUp();
```

Comment:
Call parent method first whenever possible.

Reply:
Why is it necessary? Are some requirements for it?

Reply:
@RomaKis that's necessary for readability perspective either to guarantee that your code will never access to unassigned properties.

Such programming language as Java forbid to call `super` if it's not the first call in child implementation

Reply:
Done.